packages/shared/src/utils/deprecated.test.ts

```diff
@@ -0,0 +1,201 @@
+jest.mock('./runtimeEnvironment', () => {
+  return { isDevelopmentEnvironment: jest.fn(() => true) };
+});
+
+import { deprecated, deprecatedProperty } from './deprecated';
+import { isDevelopmentEnvironment } from './runtimeEnvironment';
+
+describe('deprecated(fnName, warning)', () => {
+  let consoleWarnSpy;
+
+  beforeEach(() => {
+    consoleWarnSpy = jest.spyOn(global.console, 'warn').mockImplementation();
+  });
+  afterEach(() => {
+    consoleWarnSpy.mockRestore();
+  });
+
+  test('deprecate class method shows warning', () => {
+    class Example {
+      getSomething = (arg1?, arg2?) => {
+        deprecated('getSomething', 'Use `getSomethingElse` instead.', 'key:getSomething:1');
```

Comment:
Commenting here but that applies to all `deprecated` instances:

You're testing with the `key` provided here. It would be good to also add test cases where `key` is not given (since it's optional)

Reply:
The issue is that having a module-scope variable (as cache) causes issues in testing since the deprecations are being silenced because they are cached as displayed if we are using the same key and message. To avoid exposing the cache (to use .clear() after each test) we should either use the key to generate a different key for each deprecation or use a different message.
The only options I could think of are:
- expose `displayedWarnings` Set and `.clear()` in after each test
- use different example (eg use different method names) per each test case
- provide keys in each test case so that each deprecation is unique

What are your thoughts on that? Is there any suggestion or preference to the above?
cc: @LekoArts

Reply:
Since we want to test the codepath without a `key` the options that would rely on a `key` are not something we should do.

Use different `fnName` to keep things unique :)

Reply:
tests updated!